spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/command/ConfigCommands.java

```diff
@@ -163,11 +174,37 @@ public String target(
 		try {
 			this.targetHolder.setTarget(new Target(targetUriString, targetUsername, targetPassword, skipSslValidation));
 
-			HttpClientUtils.prepareRestTemplate(this.restTemplate,
+			HttpClientUtils.prepareRestTemplate(this.restTemplate, this.targetHolder.getTarget().getTargetUri(),
 					targetUsername, targetPassword, skipSslValidation);
 
 			this.shell.setDataFlowOperations(new DataFlowTemplate(targetHolder.getTarget().getTargetUri(), this.restTemplate));
 			this.targetHolder.getTarget().setTargetResultMessage(String.format("Successfully targeted %s", targetUriString));
+
+			final SecurityInfoResource securityInfoResource = restTemplate.getForObject(targetUriString + "/security/info", SecurityInfoResource.class);
+
+			if (securityInfoResource.isAuthenticationEnabled() && securityInfoResource.isAuthorizationEnabled()) {
+				List<CommandMarker> commandMarkersToRemove = new ArrayList<>(0);
+				for (CommandMarker commandMarker : jLineShellComponent.getSimpleParser().getCommandMarkers()) {
+					final HasRole hasRoleAnnotation = commandMarker.getClass().getAnnotation(HasRole.class);
+
+					if (hasRoleAnnotation != null && hasRoleAnnotation.value().length > 0) {
+						boolean found = false;
+						for (RoleType roleType : hasRoleAnnotation.value()) {
+							if (RoleType.VIEW.equals(roleType)) {
```

Comment:
I think the idea is not to __enforce__ security constraints (which __of course__ are enforced on the server), but rather to prevent the user from seeing things he doesn't have access to (remove clutter).